e2e-testing/__tests__/orders.test.ts

```diff
@@ -0,0 +1,387 @@
+        0,
+        undefined,
+        undefined,
+        undefined,
+        undefined,
+        undefined,
+        10,
+        undefined,
+        undefined,
+        true),
+    ]);
+    expect(ordersResponse[0]).toEqual(
+      expect.objectContaining({
+        subaccountId: SubaccountTable.uuid(DYDX_LOCAL_ADDRESS, 0),
+        clobPairId: '0',
+        side: 'BUY',
+        size: '0.001',
+        totalFilled: '0.0005',
+        price: '50000',
+        type: 'LIMIT',
+        timeInForce: 'GTT',
+        reduceOnly: false,
+        orderFlags: '64',
+        postOnly: false,
+        ticker: 'BTC-USD'
+      }),
+    );
+    expect(ordersResponse2[0]).toEqual(
+      expect.objectContaining({
+        subaccountId: SubaccountTable.uuid(DYDX_LOCAL_ADDRESS_2, 0),
+        clobPairId: '0',
+        side: 'SELL',
+        size: '0.0005',
+        totalFilled: '0.0005',
+        price: '50000',
+        type: 'LIMIT',
+        status: 'FILLED',
+        timeInForce: 'GTT',
+        reduceOnly: false,
+        orderFlags: '64',
+        postOnly: false,
+        ticker: 'BTC-USD'
+      }),
+    );
+
+    // Check API /v4/perpetualPositions endpoint
+    const [response, response2] = await Promise.all([
+      indexerClient.account.getSubaccountPerpetualPositions(DYDX_LOCAL_ADDRESS, 0),
+      indexerClient.account.getSubaccountPerpetualPositions(DYDX_LOCAL_ADDRESS_2, 0)
+    ]);
+    expect(response.positions.length).toEqual(1);
+    expect(response.positions[0]).toEqual(
+      expect.objectContaining({
+        market: 'BTC-USD',
+        status: 'OPEN',
+        side: 'LONG',
+        entryPrice: '50000',
+      }),
+    );
+    expect(response2.positions.length).toEqual(1);
+    expect(response2.positions[0]).toEqual(
+      expect.objectContaining({
+        market: 'BTC-USD',
+        status: 'OPEN',
+        side: 'SHORT',
+        entryPrice: '50000',
+      }),
+    );
+
+    // Check API /v4/orderbooks endpoint
+    const orderbooksResponse = await indexerClient.markets.getPerpetualMarketOrderbook('BTC-USD');
+    expect(orderbooksResponse).toEqual(
+      expect.objectContaining({
+        bids:[
+          {
+            price: '50000',
+            size: '0.0005',
+          }
+        ],
+        asks:[]
+      }),
+    );
+
+    // Check API /v4/candles endpoint
+    const candlesResponse = await indexerClient.markets.getPerpetualMarketCandles(
+      'BTC-USD',
+      CandleResolution.ONE_MINUTE,
+      undefined,
+      undefined,
+      1,
+    );
+    expect(candlesResponse.candles[0]).toEqual(
+      expect.objectContaining({
+        startedAt: candleStart,
+        ticker: 'BTC-USD',
+        resolution: '1MIN',
+        low: '50000',
+        high: '50000',
+        open: '50000',
+        close: '50000',
+        baseTokenVolume: '0.0005',
+        usdVolume: '25',
+        trades: 1,
+      }),
+    );
+  });
+
+
+  function connectAndValidateSocketClient(): void {
```

Comment:
is there a way to make this socket logic more generic between orders and transfer e2e tests?

Reply:
done